tests/Test.ADAL.NET.Integration/SovereignCloudsTests.cs

```diff
@@ -44,11 +78,55 @@ public void Initialize()
 
             HttpMessageHandlerFactory.AddMockHandler(new MockHttpMessageHandler
             {
+                Method = HttpMethod.Get,
```

Comment:
add another test case where, another instance discovery call is made because the authority was not already in the instance cache

Reply:
I added in a test case, but not sure if the implementation is correct.